### PR TITLE
3.0: Encrypt EBS(root volume and shared EBS) by default

### PR DESCRIPTION
**Pull request by hanwen-pcluste**
1. Change the default of encrypted to True
2. Create constants.py to share constants across tests
3. Amend test_ebs_single to test encryption by default
4. Amend test_ebs_multiple to test disabling encryption
5. Only allow queue compute settings if the scheduler is slurm
6. Only allow queue IAM if the scheduler is slurm

Signed-off-by: Hanwen <hanwenli@amazon.com>

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -38,6 +38,8 @@ per cluster.
 - Rename NodeType and tags from Master to HeadNode.
 - Remove Ganglia support.
 - Add support for associating an existing Elastic IP to the head node.
+- Encrypt EBS volumes (both the root EBS volumes and shared EBS volumes) by default. 
+  Note that if the scheduler is AWS Batch, the root volumes of the compute nodes cannot be encrypted by parallelcluster.
```
Comment (enrico-usai) [inline CHANGELOG.md:42]: nit: Encrypt root EBS volumes and shared EBS volumes by default
nit: ParallelCluster

File: cli/src/pcluster/templates/cluster_stack.py [hunk 2]
```diff
@@ -138,9 +138,10 @@ def _add_resources(self):
         # Head Node EC2 Iam Role
         self._add_role_and_policies(self.config.head_node, "HeadNode")
 
-        # Compute Nodes EC2 Iam Roles
-        for queue in self.config.scheduling.queues:
-            self._add_role_and_policies(queue, queue.name)
+        if self._condition_is_slurm():
```
Comment (enrico-usai) [inline cli/src/pcluster/templates/cluster_stack.py:141]: we should move this logic in the SlurmConstruct if it's slurm specific

File: tests/integration-tests/constants.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
+# Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
```
Comment (enrico-usai) [inline tests/integration-tests/constants.py:1]: minor: 2021

File: tests/integration-tests/tests/storage/test_ebs/test_ebs_multiple/pcluster.config.yaml [hunk 4]
```diff
@@ -6,10 +6,19 @@ HeadNode:
     SubnetId: {{ public_subnet_id }}
   Ssh:
     KeyName: {{ key_name }}
+  LocalStorage:
+    RootVolume:
+      Encrypted: false # Test turning of root volume encryption
```
Comment (enrico-usai) [inline tests/integration-tests/tests/storage/test_ebs/test_ebs_multiple/pcluster.config.yaml:11]: turning off?

File: tests/integration-tests/tests/storage/test_ebs.py [hunk 5]
```diff
@@ -110,22 +113,40 @@ def test_ebs_multiple(scheduler, pcluster_config_reader, clusters_factory, regio
...
+    _test_root_volume_encryption(cluster, os, region, scheduler, encrypted=False)
+
 
 def _get_ebs_settings_by_name(config, name):
     for shared_storage in config["SharedStorage"]:
         if shared_storage["Name"] == name:
             return shared_storage["EbsSettings"]
 
 
+def _get_root_volume_id(instance_id, os, region):
```
Comment (enrico-usai) [inline tests/integration-tests/tests/storage/test_ebs.py:141]: This seems the same function we have in `test_tag_propagation`, let's put them in common.

File: CHANGELOG.md (hunk 1 above)
Comment (hanwen-pcluste) [reply CHANGELOG.md]: Done

File: cli/src/pcluster/templates/cluster_stack.py (hunk 2 above)
Comment (hanwen-pcluste) [reply cli/src/pcluster/templates/cluster_stack.py:141]: I tried but it is not easy. To do so, we have to move `_add_role_and_policies` and related methods to `cdk_builder_utils`

File: tests/integration-tests/tests/storage/test_ebs.py (hunk 5 above)
Comment (hanwen-pcluste) [reply tests/integration-tests/tests/storage/test_ebs.py]: Good catch! Done!

**Review by enrico-usai — APPROVED** [review]

**Pull request merged**